test/k8sT/istio.go

```diff
@@ -316,12 +362,53 @@ var _ = SkipContextIf(func() bool {
 				Expect(err).To(BeNil(), "DNS entry is not ready after timeout")
 			}
 
-			By("Testing L7 filtering")
+			By("Testing Istio service configuration")
 			reviewsPodV1, err := kubectl.GetPods(helpers.DefaultNamespace, formatLabelArgument(app, reviews, version, v1)).Filter(podNameFilter)
 			Expect(err).Should(BeNil(), "Cannot get reviewsV1 pods")
+			ratingsPodV1, err := kubectl.GetPods(helpers.DefaultNamespace, formatLabelArgument(app, ratings, version, v1)).Filter(podNameFilter)
+			Expect(err).Should(BeNil(), "Cannot get ratingsV1 pods")
+			detailsPodV1, err := kubectl.GetPods(helpers.DefaultNamespace, formatLabelArgument(app, details, version, v1)).Filter(podNameFilter)
+			Expect(err).Should(BeNil(), "Cannot get detailsV1 pods")
 			productpagePodV1, err := kubectl.GetPods(helpers.DefaultNamespace, formatLabelArgument(app, productPage, version, v1)).Filter(podNameFilter)
 			Expect(err).Should(BeNil(), "Cannot get productpageV1 pods")
 
+			err = helpers.WithTimeout(func() bool {
+				allGood := true
+
+				allGood = shouldHaveService(reviewsPodV1.String(), ratings, apiPort, outbound) && allGood
+				allGood = shouldHaveService(ratingsPodV1.String(), ratings, apiPort, inbound) && allGood
+				allGood = shouldHaveService(productpagePodV1.String(), details, apiPort, outbound) && allGood
+				allGood = shouldHaveService(detailsPodV1.String(), details, apiPort, inbound) && allGood
+				allGood = shouldHaveService(productpagePodV1.String(), ratings, apiPort, outbound) && allGood
+
+				return allGood
+			}, "Istio sidecar proxies are not configured", &helpers.TimeoutConfig{Timeout: helpers.HelperTimeout})
+			Expect(err).Should(BeNil(), "Cannot configure Istio sidecar proxies")
+
+			By("Testing service local access")
```

Comment:
So this test will execute the commands inside the pod that each service runs to ensure that everything is OK? (there is no Istio involved here at all.)

Reply:
Right. Sidecar redirection in the TPROXY mode is broken in the upstream Istio 1.6.14 release. I added these tests to rule out issues with reaching the local service from the sidecar container.